tests/run-make/msvc-wholearchive/rmake.rs

```diff
@@ -0,0 +1,53 @@
+//! This is a regression test for #129020
+//! It ensures we can use `/WHOLEARCHIVE` to link a rust staticlib into DLL
+//! using the MSVC linker
+
+//@ only-msvc
+// Reason: this is testing the MSVC linker
+
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+use run_make_support::{cc, cmd, env_var, extra_c_flags, rustc};
+
+fn main() {
+    // Build the staticlib
+    rustc().crate_type("staticlib").input("static.rs").output("static.lib").run();
+    // Build an empty object to pass to the linker.
+    cc().input("c.c").output("c.obj").args(["-c"]).run();
+
+    // Find the C toolchain's linker.
+    let mut linker = PathBuf::from(env_var("CC"));
+    let linker_flavour = if linker.file_stem() == Some(OsStr::new("cl")) {
```

Comment:
Nit: there's a [`PartialEq` impl for `OsStr` and `str`](https://doc.rust-lang.org/std/ffi/struct.OsStr.html#impl-PartialEq%3Cstr%3E-for-OsStr) IIRC, so this can be

```
if linker.file_stem().is_some_and(|s| s == "cl")
```